### PR TITLE
Pass bytes to Seq from SFF parser

### PR DESCRIPTION
**Pull request by peterjc**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/SeqIO/SffIO.py [hunk 1]
```diff
@@ -1360,7 +1360,7 @@ def write_record(self, record):
         # Basics
         name = record.id.encode()
         name_len = len(name)
-        seq = record.seq.upper().encode()
+        seq = record.seq.upper().encode("ASCII")  # TODO - how to get bytes now?
```
Comment (peterjc) [inline Bio/SeqIO/SffIO.py:1363]: @mdehoon is there a clean way to get the bytes string directly from a ``Seq`` object now? Other than ``._data`` I mean.

Comment (mdehoon) [reply Bio/SeqIO/SffIO.py:1363]: Yes: `bytes(record.seq)`

Comment (peterjc) [reply Bio/SeqIO/SffIO.py:1363]: How does that work internally?

Comment (mdehoon) [reply Bio/SeqIO/SffIO.py]: `bytes` calls the `__bytes__` method of the argument.

Comment (peterjc) [reply Bio/SeqIO/SffIO.py]: And when the ``Seq`` object here doesn't have a ``__bytes__`` method, falls back on ``__getitem__``?

Comment (peterjc) [reply Bio/SeqIO/SffIO.py]: Sorry, Seq does have __bytes__ now. Must have had an old copy open.

**Review by mdehoon — APPROVED** [review]

**Pull request merged**